android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java

```diff
@@ -211,6 +211,16 @@ public Integer getFreeDiskStorage() {
     return null;
   }
 
+  @ReactMethod
+  public void isBatteryCharging(Promise p){
+    IntentFilter ifilter = new IntentFilter(Intent.ACTION_BATTERY_CHANGED);
+    Intent batteryStatus = this.reactContext.getApplicationContext().registerReceiver(null, ifilter);
+    int status = batteryStatus.getIntExtra(BatteryManager.EXTRA_STATUS, -1);
+    boolean isCharging = status == BatteryManager.BATTERY_STATUS_CHARGING ||
+                     status == BatteryManager.BATTERY_STATUS_FULL;
```

Comment:
Why would isBatteryCharging() return `true` when the battery is full ? 🤔

Reply:
Makes sense ... removing it